### PR TITLE
C-API test for ResNet50 with Passes and MKL-DNN support

### PR DESCRIPTION
**Pull request by wojtuss**
This PR is a redesigned version of https://github.com/PaddlePaddle/Paddle/pull/13369/

This patch adds a C-API test for ResNet50 with support for MKL-DNN and fuses implemented using Pass.
Also, it adds support for choosing passes to be executed in a given order.
The environment flag `FLAGS_use_mkldnn` is handled.
Different sequences of passes can be chosen in case MKL-DNN is enabled or not.
For now, only fake imagenet data is supported.

### CODE REVIEW

File: paddle/fluid/inference/api/analysis_predictor.cc [hunk 1]
```diff
@@ -75,7 +90,19 @@ bool AnalysisPredictor::Init(
...
+#endif
   OptimizeInferenceProgram();
+#ifdef PADDLE_WITH_MKLDNN
+  if (mkldnn_enabled) {
+    executor_->EnableMKLDNN(*inference_program_);
+  }
+#endif
   ctx_ = executor_->Prepare(*inference_program_, 0);
   if (config_._use_mkldnn) {
     executor_->EnableMKLDNN(*inference_program_);
```
Comment (Sand3r-) [inline paddle/fluid/inference/api/analysis_predictor.cc:108]: Shouldn't that part be removed?

Comment (wojtuss) [reply paddle/fluid/inference/api/analysis_predictor.cc:108]: True, one of them.

**Comment by wojtuss** [comment]
The code in this PR was refactored and updated into the new branch: https://github.com/wojtuss/Paddle/tree/wojtuss/resnet50-pass3.
Because I am still waiting for the "OP placement" pass implementation from Baidu, I have not updated this PR yet.

**Comment by luotao1** [comment]
This PR could be closed due to
- #13663 add resnet50 inference test.
- #13727 Enable MKL-DNN in Naive Executor and Analysis Predictor before op placement Pass.

**Pull request closed without merging**